losses/binary_losses.py

```diff
@@ -245,7 +246,7 @@ def loss(y_true: tf.Tensor, y_pred: tf.Tensor) -> tf.Tensor:
         :param y_pred: Predicted masks (tf.Tensor, shape=(<BATCH_SIZE>, <IMAGE_HEIGHT>, <IMAGE_WIDTH>, 1))
         :return: Focal loss (tf.Tensor, shape=(<BATCH_SIZE>,))
         """
-        f_loss = beta * (1 - y_pred) ** gamma * y_true * K.log(y_pred)  # β*(1-p̂)ᵞ*p*log(p̂)
+        f_loss = beta * (1 - y_pred) ** gamma * y_true * K.log(y_pred + epsilon)  # β*(1-p̂)ᵞ*p*log(p̂)
```

Comment:
This makes sense. What do you think about putting a `K.clip(y_pred, tf.keras.backend.epsilon(), 1 - tf.keras.backend.epsilon())` above instead?